plugins/by-name/compiler/default.nix

```diff
@@ -0,0 +1,19 @@
+{
+  lib,
+  ...
+}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "compiler";
+  originalName = "compiler.nvim";
+  package = "compiler-nvim";
```

Comment:
Should we be passing `callSetup = false` and/or `hasSettings = false`?

Reply:
It looked like upstream had a setup function and passes an opts table, but they don't currently have documented configuration.

Reply:
The lazy.nvim usage example includes `opts = {}`, making me think they _do_ have a setup function and maybe a settings table?

Reply:
So, I didn't want to restrict future configuration if they do start using the opts that can be provided. https://github.com/Zeioth/compiler.nvim/blob/e889774c7f2fefad60f8e6638d324223b07d2624/lua/compiler/init.lua#L9

Reply:
I came to the same conclusion when I did my own research :+1: 

The setup function does take an opts parameter, but it is currently unused internally.

I agree we should keep things as-is.